schemaregistry-avro/src/main/scala/com/microsoft/azure/schemaregistry/spark/avro/AvroDataToCatalyst.scala

```
@@ -0,0 +1,78 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.azure.schemaregistry.spark.avro
+
+import com.azure.core.util.BinaryData
+import com.azure.core.util.serializer.TypeReference
+import org.apache.avro.generic.GenericRecord
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodeGenerator, CodegenContext, ExprCode}
+import org.apache.spark.sql.catalyst.expressions.{ExpectsInputTypes, Expression, UnaryExpression}
+import org.apache.spark.sql.types._
+
+case class AvroDataToCatalyst(
+     child: Expression,
+     schemaId: String,
+     options: Map[java.lang.String, java.lang.String])
+  extends UnaryExpression with ExpectsInputTypes with Logging{
+
+  override def inputTypes: Seq[BinaryType] = Seq(BinaryType)
+
+  override lazy val dataType: DataType = {
+    val dt = SchemaConverters.toSqlType(schemaRegistryConstructor.expectedSchema).dataType;
+    dt
+  }
+
+  override def nullable: Boolean = true
+
+  @transient private lazy val schemaRegistryConstructor = SchemaRegistryConstructor.init(schemaId, options)
+
+  @transient private lazy val avroConverter = {
+    new AvroDeserializer(schemaRegistryConstructor.expectedSchema, dataType)
+  }
+
+  override def nullSafeEval(input: Any): Any = {
+    val message = new com.azure.core.models.MessageContent()
+      .setBodyAsBinaryData(BinaryData.fromBytes(input.asInstanceOf[Array[Byte]]))
+      .setContentType(("avro/binary+"+schemaRegistryConstructor.schemaId))
+    val genericRecord = schemaRegistryConstructor.serializer.deserialize(message, TypeReference.createInstance(classOf[GenericRecord]))
+    logInfo("Finished deserializing Message Content")
+
+    try {
+      logInfo("Deserializing Avro data to Catalyst data")
+      avroConverter.deserialize(genericRecord)
+    } catch {
+      case e:Throwable =>
```

1. Do we need to catch Throwable here? Throwable is a superclass that include both Error and Exception, but I don't think I see you handled anything about the Error class for example. It is ok to catch Throwable if we intend to do something about it (e.g. log details) but still, I think we should just throw back whatever we have got in that case.
2. Can we also log error in this case? In case user just ignore all exception the log should always include the error and stack for debugging purposes.

---

fixed